### PR TITLE
Automate broken link checking

### PR DESCRIPTION
**Pull request by benhyland**
## What

Added the python linkchecker tool (https://github.com/wummel/linkchecker)
to allow automatic checking for broken links, configured for internal links only at present.

Fixed broken links it discovered.

Added travis buildfile to allow this to be automated as part of our CI.
## How to review

in a local checkout, run `./check_links.sh` and verify it finds no broken links.
Deliberately break a link and verify a second run finds the broken link.
Merge this PR.
Verify the Travis build runs and finds no broken links (https://travis-ci.org/alphagov/paas-developer-docs).
## Who can review

Anyone but @benhyland

### CODE REVIEW

**Comment by Jonty** [comment]
It would be nice to squash these commits slightly before merging.

Perhaps:
- Implement linkchecker
- Fix issues found with linkchecker
- Ignore intellij

Also unsure if we should copy the same style as our other repos and have a /scripts dir along with a (slightly redundant) Makefile to execute it, it's inevitable that this repo will breed more scripts eventually!

@timmow / @dcarley, thoughts on copying layout?

**Comment by Jonty** [comment]
Oh, also can we link https://github.com/wummel/linkchecker from the commit message itself, as people are unlikely to find this PR when commit spelunking in the future.

**Comment by benhyland** [comment]
@Jonty I don't care enough about standardising on make and dir structure until we have a lot more scripts here or we are prepared to split out a shared repo for build scripts and do the work to reuse it in each project.
Happy with your other suggestions - I have a readme update anyway now @benh-gds has taken a look so I'll roll it all together and squash as you suggest.

**Comment by benhyland** [comment]
@Jonty spoke with @timmow : I will write a card to create a build scripts repo, and include this fragment in the requirements so it can be applied to our other docs as well.

Also, addressed comments and rebased as suggested.

File: README.md [hunk 1]
```diff
@@ -34,3 +35,19 @@ To preview your changes locally run:
...
+You can test manually using
+
+    ./check_links.sh
+
+The script starts a mkdocs server of its own, and kills it afterwards.
+
+The link checker will check whether internal links point to pages which actually exist.
+It won't check anchors (https://github.com/wummel/linkchecker/issues/557) and won't do much more than check well-formedness for mailto links.
+
+[Travis][] runs `./cehck_links.sh` on each commit.
```
Comment (mtekel) [inline README.md]: typo

**Comment by mtekel** [comment]
I have changed a link to loggregator architecture to docss (https://docss.cloudfoundry.org/loggregator/architecture.html), which returns 404, but I got no errors from linkchecker:

```
Statistics:
Downloaded: 538.18KB.
Content types: 43 image, 1510 text, 0 video, 0 audio, 173 application, 78 mail and 284 other.
URL lengths: min=19, max=1254, avg=45.

That's it. 2088 links in 43 URLs checked. 0 warnings found. 0 errors found.
```

Is this expected?

**Comment by benhyland** [comment]
@mtekel yes, since we only check internal links at present.
Sorry for the confusion, I updated the readme to be clearer and addressed the other comment.

**Comment by mtekel** [comment]
In that case it works as expected, merging.

**Pull request merged**